#!/usr/bin/env python
PACKAGE = "eppacher_hw2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_param_1", double_t, 0, "A double parameter",    4, -10,   10)#erstes: default value ||zweites: min value || drittes: max value
gen.add("y_param_1", double_t, 0, "A double parameter",    0, -10,   10)
gen.add("theta_goal_param_1", int_t, 0, "A double parameter",    0, 0,   360)

gen.add("x_param_2", double_t, 0, "A double parameter",    6, -10,   10)#erstes: default value ||zweites: min value || drittes: max value
gen.add("y_param_2", double_t, 0, "A double parameter",    -2, -10,   10)
gen.add("theta_goal_param_2", int_t, 0, "A double parameter",    270, 0,   360)

gen.add("x_param_3", double_t, 0, "A double parameter",    4, -10,   10)#erstes: default value ||zweites: min value || drittes: max value
gen.add("y_param_3", double_t, 0, "A double parameter",    -4, -10,   10)
gen.add("theta_goal_param_3", int_t, 0, "A double parameter",    180, 0,   360)

gen.add("x_param_4", double_t, 0, "A double parameter",    4, -10,   10)#erstes: default value ||zweites: min value || drittes: max value
gen.add("y_param_4", double_t, 0, "A double parameter",    0, -10,   10)
gen.add("theta_goal_param_4", int_t, 0, "A double parameter",    180, 0,   360)

gen.add("rho_param", double_t, 0, "A double parameter",    0.01, -1,   1)
gen.add("alpha_param", double_t, 0, "A double parameter",    0.3, -1,   1)
gen.add("beta_param", double_t, 0, "A double parameter",    0.1, -1,   1)

gen.add("turn_on_turn_off",   bool_t,   0, "A Boolean parameter",  False)


exit(gen.generate(PACKAGE, "eppacher_hw2", "dynamic_reconfigure_param_2"))